contracts/tokenfunds.js

```diff
@@ -0,0 +1,583 @@
+    && api.assert(typeof maxAmountPerDay === 'string' && api.BigNumber(maxAmountPerDay).gt(0), 'invalid maxAmountPerDay: greater than 0')) {
+    if (proposalFee) {
+      if (!api.assert(typeof proposalFee === 'object'
+        && typeof proposalFee.method === 'string' && FeeMethod.indexOf(proposalFee.method) !== -1
+        && typeof proposalFee.symbol === 'string'
+        && typeof proposalFee.amount === 'string' && api.BigNumber(proposalFee.amount).gt(0), 'invalid proposalFee')) return;
+      const feeTokenObj = await api.db.findOneInTable('tokens', 'tokens', { symbol: proposalFee.symbol });
+      if (!api.assert(feeTokenObj && api.BigNumber(proposalFee.amount).dp() <= feeTokenObj.precision, 'invalid proposalFee token or precision')) return;
+    }
+    const existingDtf = await api.db.findOne('funds', { id: fundId });
+    if (!api.assert(existingDtf, 'DTF not found')
+      || !api.assert(existingDtf.creator === api.sender || api.owner === api.sender, 'must be DTF creator')) return;
+    const payTokenObj = await api.db.findOneInTable('tokens', 'tokens', { symbol: existingDtf.payToken });
+    const voteTokenObj = await api.db.findOneInTable('tokens', 'tokens', { symbol: existingDtf.voteToken });
+    if (!api.assert(api.BigNumber(maxAmountPerDay).dp() <= payTokenObj.precision, 'maxAmountPerDay precision mismatch')
+      || !api.assert(api.BigNumber(voteThreshold).dp() <= voteTokenObj.precision, 'voteThreshold precision mismatch')) return;
+    existingDtf.voteThreshold = voteThreshold;
+    existingDtf.maxDays = maxDays;
+    existingDtf.maxAmountPerDay = maxAmountPerDay;
+    if (proposalFee) existingDtf.proposalFee = proposalFee;
+    await api.db.update('funds', existingDtf);
+
+    // burn the token creation fees
+    if (api.sender !== api.owner && api.BigNumber(dtfUpdateFee).gt(0)) {
+      await api.executeSmartContract('tokens', 'transfer', {
+        // eslint-disable-next-line no-template-curly-in-string
+        to: 'null', symbol: "'${CONSTANTS.UTILITY_TOKEN_SYMBOL}$'", quantity: dtfUpdateFee, isSignedWithActiveKey,
+      });
+    }
+    api.emit('updateFund', { id: fundId });
+  }
+};
+
+actions.setDtfActive = async (payload) => {
+  const {
+    fundId,
+    active,
+    isSignedWithActiveKey,
+  } = payload;
+
+  if (!api.assert(isSignedWithActiveKey === true, 'you must use a transaction signed with your active key')) {
+    return;
+  }
+  const dtf = await api.db.findOne('funds', { id: fundId });
+  if (api.assert(dtf, 'DTF does not exist')
+    && api.assert(dtf.creator === api.sender || api.owner === api.sender, 'must be DTF creator')) {
+    dtf.active = !!active;
+    await api.db.update('funds', dtf);
+    api.emit('setDtfActive', { id: dtf.id, active: dtf.active });
+  }
+};
+
+actions.createProposal = async (payload) => {
+  const {
+    fundId, title, startDate, endDate, amountPerDay,
+    authorperm, payout, isSignedWithActiveKey,
```

Comment:
authorperm feels to me like it's short for "author permissions".
Can you change that to something like authorpermlink, or aboutlink, to clarify the intent?

Reply:
authorperm naming came from hived, updated to authorPermlink to more suit our naming style